### PR TITLE
Refactor E2E tests

### PR DESCRIPTION
**Pull request by pranavkm**
Re-enable skipped localization tests

Fixes https://github.com/aspnet/Entropy/issues/186

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
Seems reasonable.

File: samples/Localization.StarterWeb/Models/ApplicationDbContext.cs [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore;
 
 namespace Localization.StarterWeb.Models
 {
     public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
     {
+        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
```
Comment (pranavkm) [inline samples/Localization.StarterWeb/Models/ApplicationDbContext.cs]: EF was complaining

File: test/FunctionalTests/E2ETestBase.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task WebListenerX64CoreCLRWindows()
+        {
+            return RunTestAsync(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, BasePort + 4);
+        }
+
+        [ConditionalFact(Skip = "Skipped until https://github.com/aspnet/Hosting/issues/949")]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task WebListenerX86CoreCLRWindows()
```
Comment (pranavkm) [inline test/FunctionalTests/E2ETestBase.cs]: @Tratcher should the weblistener variants just go away? Alternatively what should we do to fix it - change the project? #188

**Comment by pranavkm** [comment]
*bump*

File: samples/Localization.CustomResourceManager/Startup.cs [hunk 3]
```diff
@@ -44,6 +44,7 @@ public void ConfigureServices(IServiceCollection services)
             IStringLocalizer<YellingService> yellingLocalizer,
             IStringLocalizer<GreetingService> greetingLocalizer)
         {
+
```
Comment (ryanbrandenburg) [inline samples/Localization.CustomResourceManager/Startup.cs]: Extra new-line here.

File: test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs [hunk 4]
```diff
@@ -1,77 +1,23 @@
...
-using Microsoft.AspNetCore.Server.IntegrationTesting;
-using Microsoft.AspNetCore.Testing.xunit;
+using System.Net.Http;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace EntropyTests.BuilderTests
 {
-    public class BuilderFilteringWebTests
+    public class BuilderFilteringWebTests : E2ETestBase
```
Comment (ryanbrandenburg) [inline test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs]: Much cleaner and less boilerplate. 👍

File: test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs [hunk 5]
```diff
@@ -1,77 +1,23 @@
...
     {
-        private const string SiteName = "Builder.Filtering.Web";
-        private readonly ITestOutputHelper _output;
         public BuilderFilteringWebTests(ITestOutputHelper output)
+            : base(output, "Builder.Filtering.Web", 5500)
```
Comment (ryanbrandenburg) [inline test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs]: I suppose it's more trouble than it's worth to try making these ports unique without doing it manually.

Comment (pranavkm) [inline test/FunctionalTests/BuilderTests/BuilderFilteringWebTests.cs]: There is an API for it in Hosting (https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TestUriHelper.cs#L35), but I'm not sure how you'd make it work without locking. These tests run in parallel (and I'd like to keep it that way).

File: test/FunctionalTests/E2ETestBase.cs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+using Xunit.Abstractions;
+
+namespace EntropyTests
+{
+    public abstract class E2ETestBase
+    {
+        private string _siteName;
+        private ITestOutputHelper _output;
+
+
```
Comment (pranavkm) [inline test/FunctionalTests/E2ETestBase.cs]: Extra line.

**Pull request closed without merging**